### PR TITLE
AI Tutor: convert chat components to TypeScript 

### PR DESCRIPTION
**Pull request by Erin007**
[CT-300](https://codedotorg.atlassian.net/browse/CT-300)

Code Tools is working on converting our components (at least the new ones) to TypeScript based on this [proposal](https://docs.google.com/document/d/15KLtvSTHFOGIVwu60l1mFZscFL8miXvZHKNCKwuzf4E/edit#heading=h.a1916db8qd8j).  To support this initiative, as a mini-project for Meetings Week, the AI Tutor team is working on converting the files in `apps/src/code-studio/components/aiTutor` to TypeScript. 

This PR tackles the components related to the chat area: `userChatMessageEditor`, `chatMessage` and `chatWorkspace`. 

I've confirmed that the chat workspace still displays messages as expected with these changes. 

Follow up work will include converting `compilationTutor`, `generalChatTutor`, `validationTutor` and the `warningModal` component to TypeScript as well. [CT-301](https://codedotorg.atlassian.net/browse/CT-301)

#55937 deletes `aiTutorPanelContainer`, creates  `aiTutorFloatingActionButton` and modifies the `aiTutorPanel` component; those will be converted to TypeScript after #55937 is merged to avoid conflicts/confusion. [CT-302](https://codedotorg.atlassian.net/browse/CT-302)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by Erin007** [comment]
cc: @kobryan0619 I heard you might be interested in trying your hand at TypeScript this week. Happy to pair on one of the remaining components if you have time/interest. Just let me know and we'll set up some time!

File: apps/src/aichat/chatApi.ts [hunk 1]
```diff
@@ -1,4 +1,9 @@
-import {Role, Status, ChatCompletionMessage, PII} from './types';
+import {
+  Role,
+  Status,
+  ChatCompletionMessage,
+  PII,
+} from '@cdo/apps/aiTutor/types';
```
Comment (Erin007) [inline apps/src/aichat/chatApi.ts:6]: @sanchitmalhotra126 I'm not totally up to speed on how your team might be using or plan to use the chatApi so maybe let's connect on what we should share/split etc....and make sure we're not stepping on each other's toes!

File: apps/src/code-studio/components/aiTutor/warningModal.jsx [hunk 2]
```diff
@@ -45,8 +46,10 @@ const WarningModal = () => {
           />
         </div>
       </AccessibleDialog>
-    )
-  );
+    );
+  } else {
+    return null;
+  }
 };
```
Comment (Erin007) [inline apps/src/code-studio/components/aiTutor/warningModal.jsx:54]: @kakiha11 This doesn't convert the `warningModal` to TypeScript, but when the `chatWorkspace` was converted to TS, it was grumpy that the `warningModal` could implicitly return `false` when it's closed so I had to explicitly return `null`. Let me know if you have time to pair on converting `warningModal` to TypeScript.

File: apps/src/aiTutor/types.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  PHONE = 'phone',
+}
+
+export const PII = [Status.EMAIL, Status.ADDRESS, Status.PHONE];
+
+export enum TutorTypes {
+  COMPILATION = 'compilation',
+  VALIDATION = 'validation',
+  GENERAL_CHAT = 'general_chat',
+}
```
Comment (Erin007) [inline apps/src/aiTutor/types.ts:43]: `TutorTypes` and `AITutorInteraction` aren't used yet, but they're coming #55995 and will be useful for converting the other components, too!

File: apps/src/aichat/chatApi.ts (hunk 1 above)
Comment (sanchitmalhotra126) [reply apps/src/aichat/chatApi.ts:6]: Sounds good! We'll likely reuse some of this for upcoming Gen AI work, but development hasn't started on that yet, so this is fine for now. We can likely pull these types into a common location if/when needed.

**Review by molly-moen — APPROVED** [review]

**Pull request merged**